pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -395,7 +395,10 @@ func (ku *KubeUtil) setupKubeletApiEndpoint() error {
 			log.Debugf("Kubelet endpoint is: %s", ku.kubeletApiEndpoint)
 			return nil
 		}
-		return fmt.Errorf("unexpected status code %d on endpoint %s%s", code, ku.kubeletApiEndpoint, kubeletPodPath)
+		if code < 500 && code >= 400 {
+			return fmt.Errorf("unexpected status code %d on endpoint %s%s", code, ku.kubeletApiEndpoint, kubeletPodPath)
+		}
+		log.Info("The communication with the kubelet is unauthorized, trying to connect over HTTP")
```

Comment:
As we can't send integration easily as it's not within a check we could probably log this at a warning level to make sure the user who want to fix it is aware of it